app/src/main/java/com/infomaniak/mail/workers/DraftsActionsWorker.kt

```diff
@@ -102,7 +113,33 @@ class DraftsActionsWorker(appContext: Context, params: WorkerParameters) : BaseC
                 }
             }
 
-            if (hasRemoteException) Result.failure() else Result.success()
+            return@writeBlocking hasRemoteException
+        }
+
+        return if (isFailure) {
+            Result.failure()
+        } else {
+            updateFolderAfterScheduledDate(scheduledDates)
```

Comment:
This function is called only if the list is not empty, because `isFailure` can be `true` when the list is not empty.